tests/unit/dataSource/filtering/tests.html

```diff
@@ -979,11 +979,11 @@
 				});
 				test("Test function filter with argument fieldExpressionsOnStrings", function () {
 					var fd;
-					ds.filter([], null, null, "ID >= 1 AND ID < 3");
```

Comment:
The old tests is not valid and needed to be changed since the purpose of that last parameter is to treat the filtered field as a string field and filter it with string conditions. 
The conditions >= and < can only apply to numeric fields. 

Since the parameter was never documented, its purpose was unclear and some incorrect changes were made to allow such a setup, however some of our other controls(like the listView) depend on that last parameter to allow filtering on fields that are not necessarily string as if they were string.